static/src/js/components/SpatialDisplay/SpatialDisplay.js

```diff
@@ -76,21 +78,37 @@ class SpatialDisplay extends Component {
       shapefile
     } = this.props
 
+    this.setState({ error: '' })
```

Comment:
It seems like there should be a better way to make sure error is getting reset. IIRC, these `setState` calls stack async which I feel could lead to some unpredictable behavior. Probably worth looking into a little closer.